### PR TITLE
Support for branch name in "fin project create"

### LINKED ISSUE #1435: Use head branch or add option for branch when using a custom git url for "project create"
**Description**

Currently docksal defaults to and requires [master](https://github.com/docksal/docksal/blob/develop/bin/fin#L3398) when using a custom git url for ``project create``.

Using the HEAD branch for the custom git url or adding an option for the remote branch would allow users to start a project with a remote URL that does not contain or does not use ``master``. Github has deprecated use of the word ``master``. 

**Steps to reproduce the issue:**

1. ``fin project create``
2. Select "  Custom
    0. Custom git repository"
3. Enter project that does not contain a ``master`` branch

**Describe the results you received:**

```
Cloning into 'test-project'...
warning: Could not find remote branch master to clone.
fatal: Remote branch master not found in upstream origin
```

I'd be willing to create a PR if the maintainers would like to move on this.

### PR DESCRIPTION
**Pull request by lmakarov**
Resolves #1435 

- Added support for branch name in fin project create
- Regenerated fin help docs

Example:

```
$ fin project create --name=myd8 --repo=git@github.com:docksal/boilerplate-drupal8.git --branch=feature/circleci

Project folder:   /Users/leonid/Work/Labs/docksal/docksal/myd8
Project software: Custom git repository
Source repo:      git@github.com:docksal/boilerplate-drupal8.git
Source branch:    feature/circleci
Project URL:      http://myd8.docksal

Do you wish to proceed? [y/n]:
```

### CODE REVIEW

File: docs/content/fin/fin-help.md [hunk 1]
```diff
@@ -363,7 +365,7 @@ aliases:
 	
 	View output from containers.
 	
-	Usage: logs [options] [SERVICE...]
+	Usage: logs [options] [--] [SERVICE...]
```
Comment (lpeabody) [inline docs/content/fin/fin-help.md:368]: Does the addition of `[--]` have an impact on this feature?

File: bin/fin [hunk 2]
```diff
@@ -3376,6 +3381,7 @@ project_create ()
 	echo -e "Project folder:   ${yellow}$(pwd)/$project_name${NC}"
 	echo -e "Project software: ${yellow}${target_cms}${NC}"
```
Comment (achekulaev) [inline bin/fin:3382]:
```suggestion
	echo -e "Project software: ${yellow}${target_cms:-<unknown>}${NC}"
```

Target CMS can be unknown now

File: bin/fin [hunk 3]
```diff
@@ -3395,8 +3401,10 @@ project_create ()
 	fi
 
 	echo -e "${green}Cloning repository...${NC}"
-	run_cli git clone -b master "${target_repo}" "${project_name}" --depth=1
-	[ ! $? -eq 0 ] && _confirm "Checkout finished with errors. Do you wish to continue?"
+	run_cli git clone \
+		$( [[ "${target_branch}" != "" ]] && echo "--branch=${target_branch}" ) \
+		"${target_repo}" "${project_name}" --depth=1
```
Comment (achekulaev) [inline bin/fin:3406]:
```suggestion
		"${target_repo}" "${project_name}" --depth=1 --no-tags
```

File: bin/fin [hunk 4]
```diff
@@ -3395,8 +3401,10 @@ project_create ()
...
+	[[ $? != 0 ]] && _confirm "Checkout finished with errors. Do you wish to continue?"
```
Comment (achekulaev) [inline bin/fin:3407]:
```suggestion
	[[ $? != 0 ]] && _confirm "Checkout finished with errors. Do you wish to continue with project init?"
```

**Review by achekulaev — CHANGES_REQUESTED** [review]

File: docs/content/fin/fin-help.md (hunk 1 above)
Comment (lmakarov) [reply docs/content/fin/fin-help.md:368]: I ran  `./scripts/fin-help-generate.sh` to update `fin help` docs. 

This particular output comes from `docker-compose logs --help`, which we do not control. docker-compose version was updates, thus the change.

File: bin/fin (hunk 2 above)
Comment (lmakarov) [reply bin/fin:3382]: @achekulaev `target_cms="Custom git repository"` in the case:

```
	# Read repo from params if available
	if [[ "$repo" != "" ]]; then
		local target_repo="$repo"
		local choice="0"
	fi
```

**Pull request merged**